### PR TITLE
Update DCGM receiver to detect supported profiling fields on a given GPU model

### PR DESCRIPTION
**Pull request by LujieDuan**
Overview: 
- Modify the DCGM receiver to support GPUs that do not emit all profiling metrics the receiver wants.

Implementation:
- On the start up of the receiver, calling `dcgm.GetSupportedMetricGroups()` to get all the available profiling metrics of the current GPUs, and use that decide which fields to set watch group on.

How is this tested:
- Modify the unit tests (the `has_gpu` tests) to use golden files. The golden files can be generated for each GPU model by: `go test -tags=gpu,has_gpu  -race ./receiver/dcgmreceiver/ -update`.

### CODE REVIEW

**Review by braydonk — COMMENTED** [review]
Review is just about Go code stuff, don't have any context on the changes themselves.

File: receiver/dcgmreceiver/client.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client.go]: Updated, thanks!

File: receiver/dcgmreceiver/client.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client.go]: Done!

File: receiver/dcgmreceiver/client_gpu_test.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client_gpu_test.go:93]: This function is also used in the scraper test, so will need to export this. But I realized I do not need to export the struct `ModelSupportedFields`, so fixed that. Thanks for the comment!

File: receiver/dcgmreceiver/client_gpu_test.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client_gpu_test.go:131]: Done!

**Review by braydonk — APPROVED** [review]
All good from my side, still needs approval from someone with GPU context.

(Optional: I realized after looking again that there's a few other spots `t.Helper` may be desirable, I'll leave it up to you though)

File: receiver/dcgmreceiver/client_gpu_test.go
Comment (suffiank) [inline receiver/dcgmreceiver/client_gpu_test.go:53]: Nice! :)

**Review by suffiank — COMMENTED** [review]
LGTM ✅ I like the way you adjusted to tests to account for GPU models. Just some minor comments.

**Review by suffiank — APPROVED** [review]

File: receiver/dcgmreceiver/client.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client.go]: Done! Thanks!

File: receiver/dcgmreceiver/client.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client.go:204]: Good idea - opened an issue https://github.com/NVIDIA/go-dcgm/issues/32 to check.

File: receiver/dcgmreceiver/client.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client.go]: Done! Those are much better naming :)

File: receiver/dcgmreceiver/client_gpu_test.go
Comment (LujieDuan) [reply receiver/dcgmreceiver/client_gpu_test.go:53]: :) Thanks!

**Pull request merged**
